### PR TITLE
Move PyTorch example training pipeline to the CPU

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

### CODE REVIEW

**Comment by JanuszL** [comment]
Builds 37985.

**Comment by bhack** [comment]
I don't understand this... so it is better to pass through CPU instead of decode with gpu engine?

**Comment by JanuszL** [comment]
DALI helps you by offloading data preprocessing computations from  CPU to GPU. If your GPU has plenty of work anyway, offloading anything only slows everything down. For PyTorch and RN50 it is a bit faster to put most of the pipeline on the CPU, for RN18 it is the opposite. But it is easy to switch between those two depending on kind of network you are training, so it is up to the user to decide what works best for him.

**Comment by bhack** [comment]
But i.e. for video decoding some Nvidia cards have dedicated compute units that don't not overlap with cuda cores. So what is the behavior in this case?

**Comment by JanuszL** [comment]
@bhack - for video decoding we have such unit, for mage processing (not only JPEG decoding, but other image operations) we use cuda cores, so this shares GPU computation power together with the network training.

**Comment by JanuszL** [comment]
Just added ability to switch between GPU and CPU based pipeline in this example.

File: docs/examples/pytorch/main.py [hunk 1]
```diff
@@ -59,9 +59,10 @@
                     help='evaluate model on validation set')
 parser.add_argument('--pretrained', dest='pretrained', action='store_true',
                     help='use pre-trained model')
-
 parser.add_argument('--fp16', action='store_true',
                     help='Run model fp16 mode.')
+parser.add_argument('--daliCPU', action='store_true',
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply docs/examples/pytorch/main.py]: Done

**Comment by JanuszL** [comment]
Builds 38093

**Comment by yaysummeriscoming** [comment]
Thanks for implementing this - I had noticed there was a slight slowdown when training resnet-50 in pytorch.

**Comment by JanuszL** [comment]
@yaysummeriscoming - for RN50 it is faster with CPU pipeline, for RN18 GPU one is faster.

**Pull request merged**